lib/auth/sessions.go

```diff
@@ -103,6 +109,49 @@ func (s *Server) CreateAppSession(ctx context.Context, req types.CreateAppSessio
 	if err = s.UpsertAppSession(ctx, session); err != nil {
 		return nil, trace.Wrap(err)
 	}
+
+	// User metadata from the audit event
+	userMetadata := identity.GetUserMetadata()
+	userMetadata.User = session.GetUser()
+	userMetadata.AWSRoleARN = req.AWSRoleARN
+
+	// Record peer for the address of the requesting connection.
+	p, ok := peer.FromContext(ctx)
+	if !ok {
+		return nil, trace.Wrap(errors.New("unable to get peer from context"))
+	}
+
+	// Now that the certificate has been issued, emit a "new session created"
+	// for all events associated with this certificate.
+	appSessionStartEvent := &apievents.AppSessionStart{
```

Comment:
Also, whenever I log in to an app through tsh, I get two "User certificate issued for [rav]" events. I'm also not sure if that's the way it's supposed to work. In any case, that's probably beyond the scope of this PR as it happens on master too.